data/org.freedesktop.portal.WebExtensions.xml

```diff
@@ -0,0 +1,155 @@
+<?xml version="1.0"?>
+<!--
+ Copyright (C) 2022 Canonical Ltd
+
+ This library is free software; you can redistribute it and/or
+ modify it under the terms of the GNU Lesser General Public
+ License as published by the Free Software Foundation; either
+ version 2 of the License, or (at your option) any later version.
+
+ This library is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ Lesser General Public License for more details.
+
+ You should have received a copy of the GNU Lesser General Public
+ License along with this library. If not, see <http://www.gnu.org/licenses/>.
+-->
+
+<node name="/" xmlns:doc="http://www.freedesktop.org/dbus/1.0/doc.dtd">
+  <!--
+      org.freedesktop.portal.WebExtensions:
+      @short_description: WebExtensions portal
+
+      The WebExtensions portal allows sandboxed web browsers to start
+      native messaging hosts installed on the host system.
+
+      Accompanying documentation for Firefox's implementation is
+      available: `Native messaging for a strictly-confined Firefox
+      <https://firefox-source-docs.mozilla.org/toolkit/components/extensions/webextensions/native-messaging-portal-design.html>`_.
+
+      This documentation describes version 1 of this interface.
+  -->
+  <interface name="org.freedesktop.portal.WebExtensions">
+    <!--
+        CreateSession:
+        @options: Vardict with optional further information
+        @session_handle: Object path for the #org.freedesktop.portal.Session created by this call.
+
+        Create a web extensions session. A successfully created
+        session can at any time be closed using
+        org.freedesktop.portal.Session::Close, or may at any time be
+        closed by the portal implementation, which will be signalled
+        via org.freedesktop.portal.Session::Closed.
+
+        To close a session, the browser should:
+
+        1. close the process's stdin/stdout/stderr file descriptors
```

Comment:
Firefox doesn't close stdout/stderr, only stdin. It is possible for the native application to continue writing to stderr which could help with debugging if desired. Here is a suggestion that is more accurate.

```suggestion
        1. close the process's stdin file descriptors, optionally also stdout/stderr.
```